### PR TITLE
Bumping up the Cloud libraries bom and spanner java client libraries version

### PR DESCRIPTION
**Pull request by darshan-sj**
Bumping up the libraries bom and spanner java client libraries version.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by darshan-sj** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by darshan-sj** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by Abacn** [comment]
Thanks, BOM dep update also need to sync with a few dependencies defined in BeamModulePlugin. Usually we update BOM dep once per release, the last time was #25790

**Comment by darshan-sj** [comment]
Linkage check tool log -- https://gist.github.com/darshan-sj/43f001e8d51dbc0297bc0b754abff8b5

**Review by suztomo — CHANGES_REQUESTED** [review]
Can you update GAX version? https://github.com/googleapis/java-cloud-bom/releases/tag/v26.11.0

> GAX: 2.23.3


Line 489


It's good that linkage checker tool is not reporting failures. Thank you for checking.

**Comment by darshan-sj** [comment]
done

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -631,7 +631,7 @@ class BeamModulePlugin implements Plugin<Project> {
         // The release notes shows the versions set by the BOM:
         // https://github.com/googleapis/java-cloud-bom/releases/tag/v26.10.0
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:632]: Can you replace any occurrence of "26.10.0" with "26.11.0"?

**Review by suztomo — CHANGES_REQUESTED** [review]

**Comment by darshan-sj** [comment]
Run Java_Spark3_Versions PreCommit

**Review by suztomo — APPROVED** [review]

**Comment by darshan-sj** [comment]
Run Java PreCommit

**Comment by darshan-sj** [comment]
Run Java PreCommit

**Comment by darshan-sj** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by darshan-sj** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by darshan-sj** [comment]
Run Java_PVR_Flink_Batch PreCommit

**Comment by darshan-sj** [comment]
Run Java_Spark3_Versions PreCommit

**Comment by suztomo** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by suztomo** [comment]
Run Java_Spark3_Versions PreCommit

**Comment by suztomo** [comment]
4 skipped and 62 successful checks.

**Pull request merged**